tests/_simulators/test_get_marginal_fock_probabilities.py

```diff
@@ -0,0 +1,282 @@
+
+        pq.Q(0) | pq.Squeezing(r=0.1)
+        pq.Q(1) | pq.Squeezing(r=0.05)
+
+    simulator = SimulatorClass(d=2, config=pq.Config(cutoff=4))
+    state = simulator.execute(program).state
+
+    # Get marginal probabilities for all modes
+    marginal_probs_all = state.get_marginal_fock_probabilities(modes=(0, 1))
+    
+    # Get full state probabilities
+    full_probs = state.fock_probabilities
+
+    # They should be equal (within numerical precision)
+    assert np.allclose(marginal_probs_all, full_probs)
+
+
+@pytest.mark.parametrize(
+    "SimulatorClass",
+    (
+        pq.GaussianSimulator,
+        pq.PureFockSimulator,
+        pq.FockSimulator,
+    ),
+)
+def test_get_marginal_fock_probabilities_consistency_with_reduced_state(SimulatorClass):
+    """Test that marginal probabilities are consistent with reduced state probabilities."""
+    
+    with pq.Program() as program:
+        pq.Q(all) | pq.Vacuum()
+
+        pq.Q(0) | pq.Squeezing(r=0.2)
+        pq.Q(1) | pq.Squeezing(r=0.1)
+        pq.Q(2) | pq.Squeezing(r=0.15)
+        pq.Q(0, 1) | pq.Beamsplitter(theta=np.pi / 6)
+        pq.Q(1, 2) | pq.Beamsplitter(theta=np.pi / 4)
+
+    simulator = SimulatorClass(d=3, config=pq.Config(cutoff=5))
+    state = simulator.execute(program).state
+
+    # Get marginal probabilities using the method
+    marginal_probs = state.get_marginal_fock_probabilities(modes=(0, 2))
+    
+    # Get probabilities from reduced state
+    reduced_state = state.reduced(modes=(0, 2))
+    reduced_probs = reduced_state.fock_probabilities
+
+    # They should be equal
+    assert np.allclose(marginal_probs, reduced_probs)
+
+
+@pytest.mark.parametrize(
+    "SimulatorClass",
+    (
+        pq.GaussianSimulator,
+        pq.PureFockSimulator,
+        pq.FockSimulator,
+    ),
+)
+def test_get_marginal_fock_probabilities_with_displacement(SimulatorClass):
+    """Test marginal probabilities with displacement operations."""
+    
+    with pq.Program() as program:
+        pq.Q(all) | pq.Vacuum()
+
+        pq.Q(0) | pq.Displacement(r = 0, phi=0.5)
+        pq.Q(1) | pq.Displacement(r = 0, phi=0.3j)
+        pq.Q(0, 1) | pq.Beamsplitter(theta=np.pi / 3)
+
+    simulator = SimulatorClass(d=2, config=pq.Config(cutoff=6))
+    state = simulator.execute(program).state
+
+    # Get marginal probabilities for each mode
+    marginal_probs_0 = state.get_marginal_fock_probabilities(modes=(0,))
+    marginal_probs_1 = state.get_marginal_fock_probabilities(modes=(1,))
+
+    # Check basic properties
+    assert np.isclose(np.sum(marginal_probs_0), 1.0,atol=1e-02)
+    assert np.isclose(np.sum(marginal_probs_1), 1.0,atol=1e-02)
+    assert np.all(marginal_probs_0 >= 0)
+    assert np.all(marginal_probs_1 >= 0)
+
+
+@pytest.mark.parametrize(
+    "SimulatorClass",
+    (
+        pq.GaussianSimulator,
+        pq.PureFockSimulator,
+        pq.FockSimulator,
+    ),
+)
+def test_get_marginal_fock_probabilities_mode_ordering(SimulatorClass):
```

Comment:
I don't understand this, as it doesn't check if the two probability distributions are equivalent.